pkg/operator/ceph/webhook-config.go

```diff
@@ -0,0 +1,237 @@
+/*
+Copyright 2022 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package operator
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	api "github.com/jetstack/cert-manager/pkg/apis/certmanager/v1"
+	v1 "github.com/jetstack/cert-manager/pkg/apis/meta/v1"
+	"github.com/rook/rook/pkg/clusterd"
+	admv1 "k8s.io/api/admissionregistration/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	issuerName        = "selfsigned-issuer"
+	certificateName   = "rook-admission-controller-cert"
+	webhookConfigName = "rook-ceph-webhook"
+)
+
+func createIssuer(ctx context.Context, clusterdContext *clusterd.Context) (*api.Issuer, error) {
+	logger.Infof("Fetching Issuer %s/%s.", namespace, issuerName)
+	issuer, err := clusterdContext.Admission.Issuers(namespace).Get(ctx, issuerName, metav1.GetOptions{})
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			logger.Infof("Creating Issuer %s/%s.", namespace, issuerName)
+			issuer = &api.Issuer{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      issuerName,
+					Namespace: namespace,
+				},
+				Spec: api.IssuerSpec{
+					IssuerConfig: api.IssuerConfig{
+						SelfSigned: &api.SelfSignedIssuer{},
+					},
+				},
+			}
+			issuers, _ := clusterdContext.Admission.Issuers(namespace).Create(ctx, issuer, metav1.CreateOptions{})
+			sleepTime := 1
+			attempts := 10
+			for i := 0; i < attempts; i++ {
+				for j := range issuers.Status.Conditions {
+					if issuers.Status.Conditions[j].Status == v1.ConditionStatus(api.IssuerConditionReady) {
+						return issuers, nil
+					}
+					logger.Debugf("webhook config %q status=%+v", issuerName, issuers.Status.Conditions[j].Status)
+				}
+				time.Sleep(time.Duration(sleepTime) * time.Second)
+				logger.Infof("waiting for webhook config %q to be in ready status", issuerName)
+			}
+			return issuers, nil
+		}
+
```

Comment:
If it's an err, but not the NotFound error, we still return nil? Seems like we should at least log the error here.